File: examples/principal.cdk/principal.cdk.py
import json
from cdk_iam_floyd import statement
import aws_cdk.aws_iam as iam

s = (
    # doc-start
    statement.Sts() \
        .allow() \
        .to_assume_role() \
        .for_cdk_principal(iam.ServicePrincipal('sns.amazonaws.com'),
                           iam.ServicePrincipal('lambda.amazonaws.com'))
    # doc-end
).to_json()

pretty = json.dumps(s, indent=4)
print(pretty)

File: examples/principal.cdk/principal.cdk.ts
import { aws_iam as iam } from 'aws-cdk-lib';

import { out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.Sts()
      .allow()
      .toAssumeRole()
      .forCdkPrincipal(
        new iam.ServicePrincipal('sns.amazonaws.com'),
        new iam.ServicePrincipal('lambda.amazonaws.com'),
      )
    // doc-end
  );
}
const s = [getStatement()];
out(s);
